File: pixel-lab-backend/utils/result.js
/**
 * 【文件路径】utils/result.js
 * 【文件功能说明】统一接口返回格式封装
 * - 成功返回
 * - 失败返回
 * - 统一格式：{ code, msg, data }
 */

/**
 * 成功响应
 * @param {object} res - Express response 对象
 * @param {*} data - 返回数据
 * @param {string} msg - 提示信息
 * @param {number} code - 状态码
 */
const success = (res, data = null, msg = '操作成功', code = 200) => {
  res.json({
    code,
    msg,
    data
  })
}

/**
 * 失败响应
 * @param {object} res - Express response 对象
 * @param {string} msg - 错误信息
 * @param {number} code - 错误码
 * @param {*} data - 额外数据
 */
const error = (res, msg = '操作失败', code = 400, data = null) => {
  res.status(code >= 400 && code < 600 ? code : 400).json({
    code,
    msg,
    data
  })
}

module.exports = {
  success,
  error
}

File: pixel-lab-java-backend/src/main/java/com/pixellab/util/Result.java
package com.pixellab.util;

import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.util.LinkedHashMap;
import java.util.Map;

public final class Result {
  private Result() {
  }

  public static void ok(HttpServletResponse response, Object data) throws IOException {
    write(response, HttpServletResponse.SC_OK, 200, "success", data);
  }

  public static void ok(HttpServletResponse response, String msg, Object data) throws IOException {
    write(response, HttpServletResponse.SC_OK, 200, msg, data);
  }

  public static void badRequest(HttpServletResponse response, String msg) throws IOException {
    write(response, HttpServletResponse.SC_BAD_REQUEST, 400, msg, null);
  }

  public static void unauthorized(HttpServletResponse response, String msg) throws IOException {
    write(response, HttpServletResponse.SC_UNAUTHORIZED, 401, msg, null);
  }

  public static void forbidden(HttpServletResponse response, String msg) throws IOException {
    write(response, HttpServletResponse.SC_FORBIDDEN, 403, msg, null);
  }

  public static void notFound(HttpServletResponse response, String msg) throws IOException {
    write(response, HttpServletResponse.SC_NOT_FOUND, 404, msg, null);
  }

  public static void serverError(HttpServletResponse response, String msg) throws IOException {
    write(response, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, 500, msg, null);
  }

  public static void unavailable(HttpServletResponse response, String msg) throws IOException {
    write(response, HttpServletResponse.SC_SERVICE_UNAVAILABLE, 503, msg, null);
  }

  public static void write(HttpServletResponse response, int httpStatus, int code, String msg, Object data) throws IOException {
    response.setStatus(httpStatus);
    response.setCharacterEncoding("UTF-8");
    response.setContentType("application/json;charset=UTF-8");

    Map<String, Object> body = new LinkedHashMap<>();
    body.put("code", code);
    body.put("msg", msg);
    body.put("data", data);
    response.getWriter().write(JsonUtil.stringify(body));
  }
}
